### ISSUE TITLE
Datepicker is not using custom dateFormat from locale

### ISSUE THREAD

**Issue by richard-schiemenz** [ant-design/ant-design#42119]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-datepicker-bug-jfgdb2?file=/index.js)

### Steps to reproduce

Wrap DatePicker in ConfigProvider with locale and dateFormat for DatePicker set.  
Select a date in the DatePicker.

### What is expected?

The date is formatted with the format specified in the locale.

### What is actually happening?

The date is formatted with the default format.

| Environment | Info |
| --- | --- |
| antd | 5.4.6 |
| React | 9.6.2 |
| System | macOS 13.3.1 |
| Browser | Chrome 112.0.5615.137 |

---
I need to set the format of the date displayed based on the locale as I have multiple locales. I have tested modifying other values in the locale object and it worked as expected. Only modifying the dateFormat seems not to be working.  
Other issues show a solution by setting the format in the DatePicker to "L" but this is a fix using moment and I want to stay with the now default dayjs.

**Comment by BoyYangzai** [comment]
You need to introduce the `dayjs` language file in advance to use it properly
the same as https://github.com/ant-design/ant-design/issues/42087

**Comment by richard-schiemenz** [comment]
I don't think this is the same issue, as there the problem lied in translating written days of the week. I had a similar issue and fixed it by importing the language files. 
Here however the formatting is not applied. Which in the example doesn't use any written out weekdays but just the digits.

**Comment by BoyYangzai** [comment]
@richard-schiemenz 
I'm sorry, I couldn't find the source code when I was looking through it before. 
This may be a previous usage, I'm sorry.

**Comment by richard-schiemenz** [comment]
@BoyYangzai  Ok, thank you.  
Maybe it should be removed from the example if it is no longer supported. As this example is linked to on the official documentation [DatePicker Localization](https://ant.design/components/date-picker#localization).  
I updated the code sandbox to show the implementation with the locale directly input to the DatePicker as shown in the first example. It also doesn't apply the format.

**Comment by Wxh16144** [comment]
I think it is indeed a bug.

**Comment by Wxh16144** [comment]
Co-authored @any1024

**Comment by Rafael-Martins** [comment]
Hey guys, looking at it to see it if I can contribute with a pr!

**Comment by any1024** [comment]
> Hey guys, looking at it to see it if I can contribute with a pr!
This is caused by the refactoring of ` rc-picker`. If you are interested, you can contribute with a pr.

**Comment by Rafael-Martins** [comment]
> > Hey guys, looking at it to see it if I can contribute with a pr!
> > This is caused by the refactoring of ` rc-picker`. If you are interested, you can contribute with a pr.

Oh cool, I was fixing my local environment. I will try to fix it today!

**Comment by Rafael-Martins** [comment]
I opened a pr with some solution, we can discuss about it in there!

**Comment by yoyo837** [comment]
Duplicate of #23902

**Comment by Josyann** [comment]
Hey, I'm encountering the same issue.
For the provider to work, the only way I've found to fix it for now it to rebuild the locales as you did but with `fieldDateFormat` and `fieldDateTimeFormat`, with the deprecated `dateFormat` and  `dateTimeFormat` values like this:
I don't have to pass any props to the DatePicker, it's only working with the `ConfigProvider`.

```typescript
const buildAntdLocales = (locales: Record<string, Locale>) => {
  return Object.entries(locales).reduce((acc, [key, value]) => {
    return {
      ...acc,
      [key]: {
        ...value,
        DatePicker: value?.DatePicker
          ? {
              ...value.DatePicker,
              lang: {
                ...value.DatePicker.lang,
                fieldDateFormat: value.DatePicker.lang.dateFormat,
                fieldDateTimeFormat: value.DatePicker.lang.dateTimeFormat,
              },
            }
          : undefined,
      },
    }
  }, {})
}

const antdLocales: Record<string, Locale> = buildAntdLocales({
  en: en_US,
  fr: fr_FR,
  es: es_ES,
})
```

Is it still supposed to be a bug? Or am I missing something?
Thanks

**Comment by DiFuks** [comment]
@yoyo837 Any updates? I'm still experiencing the issue

**Comment by Wxh16144** [comment]
> Hey, I'm encountering the same issue. For the provider to work, the only way I've found to fix it for now it to rebuild the locales as you did but with `fieldDateFormat` and `fieldDateTimeFormat`, with the deprecated `dateFormat` and `dateTimeFormat` values like this: I don't have to pass any props to the DatePicker, it's only working with the `ConfigProvider`.
> 
> const buildAntdLocales = (locales: Record<string, Locale>) => {
>   return Object.entries(locales).reduce((acc, [key, value]) => {
>     return {
>       ...acc,
>       [key]: {
>         ...value,
>         DatePicker: value?.DatePicker
>           ? {
>               ...value.DatePicker,
>               lang: {
>                 ...value.DatePicker.lang,
>                 fieldDateFormat: value.DatePicker.lang.dateFormat,
>                 fieldDateTimeFormat: value.DatePicker.lang.dateTimeFormat,
>               },
>             }
>           : undefined,
>       },
>     }
>   }, {})
> }
> 
> const antdLocales: Record<string, Locale> = buildAntdLocales({
>   en: en_US,
>   fr: fr_FR,
>   es: es_ES,
> })
> Is it still supposed to be a bug? Or am I missing something? Thanks

I would like to add a few points, the solution @Josyann replied to is only applicable to versions `5.23.4` and earlier. If `5.24.0` and later versions are used, this solution is no longer applicable.

For a specific explanation, please see here: https://github.com/ant-design/ant-design/discussions/53011.

In short, rc-picker has left some outdated formatting configurations. After two major overhauls, developers decided not to continue using the previous fields and [have deleted them](https://github.com/react-component/picker/pull/917).

If you need to customize the formatting configuration, please submit [a PR](https://github.com/react-component/picker/compare) for it to take effect in v6.

**Comment by afc163** [comment]
See https://github.com/ant-design/ant-design/issues/47063#issuecomment-1987812256

```diff
  const datePickerEn = {
    ...datePickerEnUS,
    lang: {
      ...datePickerEnUS.lang,
+     fieldDateFormat: "DD.MM.YYYY",
    },
  };
```
